packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -117,6 +118,43 @@ exports.sourceNodes = async (
     pluginConfig,
   })
 
+  createTypes(`
+    interface ContentfulEntry @nodeInterface {
```

Comment:
Note to myself:

Ensure we don't break when users use the following names for their content types:

* Reference
* Entry
* Asset